### ISSUE TITLE
Seek video on iOS with react-native-reanimated cause randomly crash at launch and in release mode only

### ISSUE THREAD

**Issue by Deodes** [TheWidlarzGroup/react-native-video#2556]
# Bug

When the app launch first time and we're trying to seek video, the app can(not always) crash with no error. 

Debugging with sentry show:

`C++ Exception RNSScreen N5folly4json11print_errorE`

Debugging from xCode console show:

`libc++abi: terminating with uncaught exception of type folly::json::print_error: folly::toJson: JSON object value was an INF when serializing value at "playableDuration"`

When not trying to seek video, or remove **react-native-reanimated** app stops crashing.
## Platform

Which player are you experiencing the problem on:
* iOS

## Environment info

React native info output:

```bash
System:
    OS: macOS 11.6.1
    CPU: (8) arm64 Apple M1
    Memory: 302.91 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - /usr/local/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.1.2 - /usr/local/bin/npm
    Watchman: 2021.12.06.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 29.0.2, 30.0.2, 31.0.0
      System Images: android-29 | ARM 64 v8a, android-29 | Intel x86 Atom_64, android-29 | Google APIs ARM 64 v8a, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play ARM 64 v8a, android-31 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.65.1 => 0.65.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: x.x.x

## Steps To Reproduce

1. Isntall **react-native-video** and **react-native-reanimated v2.3.1**
1. Launch app `react-native run-ios`
2. On App.js mount and video loaded call `this.videoRef?.seek(x)`

## Expected behaviour

1. To not catch crash, especially with no errors

## Reproducible sample code

```
onLoad() {
     this.videoPlayerRef.seek(...);
  }

<Video
     {...props}               
    onLoad={onLoad}
/>
```

**Comment by pastean** [comment]
Confirming, we're using a library that uses react-native-reanimated and this happens, not sure how it's related to this though.
```
System:
    OS: macOS 11.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 337.40 MB / 16.00 GB
    Shell: 3.1.2 - /usr/local/bin/fish
  Binaries:
    Node: 15.14.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.7.6 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 15.0, DriverKit 20.4, macOS 11.3, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2
      System Images: android-28 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.0/13A233 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_265 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.3 => 0.66.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by antoinemaillard** [comment]
We've been having the very same issue for a few weeks now and we tried to investigate.
It's a bit strange since we didn't update react-native-video (nor react-native-reanimated) but we switched to XCode 13.

The final bug is in the FB folly lib that doesn't like Infinite values for numbers and the underlying issue seems to be the `playableDuration` field that is sometimes indeed infinite.

We just tried this workaround in `RCTVideo.m` to prevent `calculatePlayableDuration` function from returning an infinite value... we'll keep you updated here if it works.

```
--- a/ios/Video/RCTVideo.m
+++ b/ios/Video/RCTVideo.m
@@ -311,9 +311,12 @@ - (NSNumber *)calculatePlayableDuration
         *stop = YES;
       }
     }];
-    Float64 playableDuration = CMTimeGetSeconds(CMTimeRangeGetEnd(effectiveTimeRange));
-    if (playableDuration > 0) {
-      return [NSNumber numberWithFloat:playableDuration];
+    CMTime endTime = CMTimeRangeGetEnd(effectiveTimeRange);
+    if (CMTIME_IS_VALID(endTime) && !CMTIME_IS_POSITIVE_INFINITY(endTime)) {
+      Float64 playableDuration = CMTimeGetSeconds(endTime);
+      if (playableDuration > 0) {
+        return [NSNumber numberWithFloat:playableDuration];
+      }
     }
   }
   return [NSNumber numberWithInteger:0];
```

**Comment by antoinemaillard** [comment]
Oh and actually, I'm not 100% sure of my analysis but the `__block` variable defined [here](https://github.com/react-native-video/react-native-video/blob/master/ios/Video/RCTVideo.m#L306) is not initialized. It seeems to be fine because legacy compilers used to unintentionally zero-initiliaze such variables but [Apple clearly states](https://opensource.apple.com/source/lldb/lldb-159/llvm/tools/clang/www/compatibility.html#block-variable-initialization) that this it not the standard behavior (and clang doesn't do so):
> By an accident of implementation, GCC and llvm-gcc unintentionally always zero initialized __block variables. However, any program which depends on this behavior is relying on unspecified compiler behavior. Programs must explicitly initialize all local block variables before they are used, as with other local variables.
> Clang does not zero initialize local block variables, and programs which rely on such behavior will most likely break when built with Clang.


It could be that the clang update that comes with XCode 13 no longer adopts this "zero-init" behavior and thus the `effectiveTimeRange` variable is left in a uninit'd/random state. If the video has no range at all, it can lead to very strange behavior.

**Comment by pastean** [comment]
Confirming that the issue is in the calculatePlayableDuration function, we've removed all the code (function + call) a couple of days ago and it just got through QA.

**Comment by orange-brother** [comment]
Confirming that this issue is still exist in react-native-video 5.2 and react 0.65.

**Comment by Aaron46** [comment]
I have the same issue. Wrapping the self.onVideoProgress call with a try-catch in 
react-native-video/ios/Video/RCTVideo.m seems to resolve the issue

(I use [patch-package](https://github.com/ds300/patch-package) (extensively...) for these changes)

```  if( currentTimeSecs >= 0 && self.onVideoProgress) {
    @try{
      self.onVideoProgress(@{
                            @"currentTime": [NSNumber numberWithFloat:CMTimeGetSeconds(currentTime)],
                            @"playableDuration": [self calculatePlayableDuration],
                            @"atValue": [NSNumber numberWithLongLong:currentTime.value],
                            @"atTimescale": [NSNumber numberWithInt:currentTime.timescale],
                            @"currentPlaybackTime": [NSNumber numberWithLongLong:[@(floor([currentPlaybackTime timeIntervalSince1970] * 1000)) longLongValue]],
                            @"target": self.reactTag,
                            @"seekableDuration": [self calculateSeekableDuration],
                            });
      }@catch(id anException){
    }
  }

**Comment by yedidyak** [comment]
@antoinemaillard Hi, did the patch you posted above help in the end? Thanks!

**Comment by antoinemaillard** [comment]
Yes @yedidyak , for us it worked (no more crashes)

**Comment by freeboub** [comment]
Can someone open a PR with the fix ?

**Comment by antoinerousseau** [comment]
I'm having the same crash

**Comment by freeboub** [comment]
Can you open a pull request with this patch please?

**Comment by freeboub** [comment]
This shall be fixed on V6.0.0-alpha*.
Let me know if we need to reopen this ticket
